src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -474,7 +474,9 @@ def _train_epoch(self, context: PhaseContext, silent_mode: bool = False) -> tupl
                 if self.pre_prediction_callback is not None:
                     inputs, targets = self.pre_prediction_callback(inputs, targets, batch_idx)
 
-                context.update_context(batch_idx=batch_idx, inputs=inputs, target=targets, **additional_batch_items)
+                context.update_context(
+                    batch_idx=batch_idx, inputs=inputs, target=targets, additional_batch_items=additional_batch_items, **additional_batch_items
+                )
                 self.phase_callback_handler.on_train_batch_start(context)
```

Comment:
I'm not 100% in love with the idea of having a magical dict (`additional_batch_items `) that is passed throughout the context but I do agree it is much cleaner than the current logic where the `**additional_batch_items` are all considered like class attributes, without any explicitness.
And technically, this is what we already do with the `sg_trainer_utils.unpack_batch_items` and the collate functions, so it's not really an issue.

So I'm good with what you propose, and I do think we should stick to it, but an alternative could be to introduce a dataclass `AdditionalBatchItems` which would be different for different collate function. In this case, we would make the contract much more explicit. But that's extra work and not really in the scope of this PR

Reply:
I generally agree. Maybe at SG 4.0 we rewrite that part completely 💅